code/datums/ai/basic_mobs/basic_ai_behaviors/stop_and_stare.dm

```diff
@@ -7,9 +7,10 @@
 	var/atom/movable/target = controller.blackboard[target_key]
 	return ismovable(target) && isturf(target.loc) && ismob(controller.pawn)
 
+/datum/ai_behavior/stop_and_stare/get_cooldown(datum/ai_controller/cooldown_for)
+	return cooldown_for.blackboard[BB_STATIONARY_COOLDOWN]
+
 /datum/ai_behavior/stop_and_stare/perform(seconds_per_tick, datum/ai_controller/controller, target_key)
-	// i don't really like doing this but we wanna make sure that the cooldown is pertinent to what we need for this specific controller before we invoke parent
-	action_cooldown = controller.blackboard[BB_STATIONARY_COOLDOWN]
 	. = ..()
```

Comment:
fixing this junky code pattern (and the whole signal stuff) makes it worth it. i think as more and more ai controllers need to exist we're probably going to arms-race to streamline performance as much as possible so may as well get some groundwork in early before silly patterns like this (my bad) get _really_ settled in and impede that progress.